label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -27,29 +37,42 @@
 
 public class LabelApplicationRuleConfigServiceImpl
     extends LabelApplicationRuleConfigServiceGrpc.LabelApplicationRuleConfigServiceImplBase {
-  static final String LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG =
+  private static final String LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG =
       "label.application.rule.config.service";
-  static final String MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT =
+  private static final String MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT =
       "max.dynamic.label.application.rules.per.tenant";
-  static final int DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT = 100;
+  private static final String SYSTEM_LABEL_APPLICATION_RULES = "system.label.application.rules";
+  private static final int DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT = 100;
+
   private final IdentifiedObjectStore<LabelApplicationRule> labelApplicationRuleStore;
   private final LabelApplicationRuleValidator requestValidator;
   private final int maxDynamicLabelApplicationRulesAllowed;
+  private final List<LabelApplicationRule> systemLabelApplicationRules;
+  private final Map<String, LabelApplicationRule> systemLabelApplicationRulesMap;
 
   public LabelApplicationRuleConfigServiceImpl(
       Channel configChannel, Config config, ConfigChangeEventGenerator configChangeEventGenerator) {
-    int maxDynamicRules = DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT;
-    if (config.hasPath(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG)) {
-      Config labelApplicationRuleConfig =
-          config.getConfig(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG);
-      if (labelApplicationRuleConfig.hasPath(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT)) {
-        maxDynamicRules =
-            labelApplicationRuleConfig.getInt(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT);
-      }
+    Config labelApplicationRuleConfig =
+        config.hasPath(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG)
```

Comment:
hasPath checks aren't necessary - just put an empty array as a default value in application.conf (which is less code and makes the config easier to understand anyway). Same goes for the other configs, although they're pre-existing.

Reply:
Keeping it as is. May require changes across multiple application.conf